operate/qa/integration-tests/src/test/java/io/camunda/operate/schema/SchemaManagerIT.java

```diff
@@ -0,0 +1,178 @@
+    mockTenantResponse();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    schemaManager.deleteIndicesFor(idxName("index-*"));
+  }
+
+  private String idxName(final String name) {
+    return indexPrefix + "-" + name;
+  }
+
+  @Test // OPE-1311
+  public void testCheckAndUpdateIndices() throws Exception {
+
+    final int expectedNumberOfReplicas = 5;
+    if (DatabaseInfo.isOpensearch()) {
+      operateProperties.getOpensearch().setNumberOfReplicas(expectedNumberOfReplicas);
+    } else {
+      operateProperties.getElasticsearch().setNumberOfReplicas(expectedNumberOfReplicas);
+    }
+    /// Old version -> before migration
+    // create index
+    createIndex(idxName("index-1.2.3_"), List.of(Map.of("test_name1", "test_value1")));
+    createIndex(idxName("index-2.3.4_"), List.of(Map.of("test_name2", "test_value2")));
+    createIndex(idxName("index-3.4.5_"), List.of(Map.of("test_name3", "test_value3")));
+
+    // set reindex settings
+    schemaManager.setIndexSettingsFor(
+        Map.of(
+            NUMBERS_OF_REPLICA, NO_REPLICA,
+            REFRESH_INTERVAL, NO_REFRESH),
+        idxName("index-1.2.3_"));
+    schemaManager.setIndexSettingsFor(
+        Map.of(REFRESH_INTERVAL, NO_REFRESH), idxName("index-2.3.4_"));
+    schemaManager.setIndexSettingsFor(
+        Map.of(NUMBERS_OF_REPLICA, "5", REFRESH_INTERVAL, NO_REFRESH), idxName("index-3.4.5_"));
+
+    // update number of replicas for each index
+    schemaManager.checkAndUpdateIndices();
+
+    final Map<String, String> reindexSettings1 =
+        schemaManager.getIndexSettingsFor(
+            idxName("index-1.2.3_"), NUMBERS_OF_REPLICA, REFRESH_INTERVAL);
+    assertThat(reindexSettings1)
+        .containsEntry(
+            NUMBERS_OF_REPLICA,
+            DatabaseInfo.isOpensearch() ? null : String.valueOf(expectedNumberOfReplicas))
+        .containsEntry(REFRESH_INTERVAL, DatabaseInfo.isOpensearch() ? null : NO_REFRESH);
+
+    final Map<String, String> reindexSettings2 =
+        schemaManager.getIndexSettingsFor(
+            idxName("index-2.3.4_"), NUMBERS_OF_REPLICA, REFRESH_INTERVAL);
+    assertThat(reindexSettings2)
+        .containsEntry(
+            NUMBERS_OF_REPLICA,
+            DatabaseInfo.isOpensearch() ? null : String.valueOf(expectedNumberOfReplicas))
+        .containsEntry(REFRESH_INTERVAL, DatabaseInfo.isOpensearch() ? null : NO_REFRESH);
+
+    final Map<String, String> reindexSettings3 =
+        schemaManager.getIndexSettingsFor(
+            idxName("index-3.4.5_"), NUMBERS_OF_REPLICA, REFRESH_INTERVAL);
+    assertThat(reindexSettings3)
+        .containsEntry(
+            NUMBERS_OF_REPLICA,
+            DatabaseInfo.isOpensearch() ? null : String.valueOf(expectedNumberOfReplicas))
+        .containsEntry(REFRESH_INTERVAL, DatabaseInfo.isOpensearch() ? null : NO_REFRESH);
+    // Migrator uses this
+    assertThat(schemaManager.getOrDefaultNumbersOfReplica(idxName("index-1.2.3_"), "5"))
```

Comment:
We should use a default value here that causes the test to fail if it's returned. Default value implies we couldn't fetch the number of replicas and that feels like it's an error condition in this test